lms/djangoapps/shoppingcart/views.py

```diff
@@ -820,20 +823,89 @@ def show_receipt(request, ordernum):
     Displays a receipt for a particular order.
     404 if order is not yet purchased or request.user != order.user
     """
+    is_json_request = 'application/json' in request.META.get('HTTP_ACCEPT', "")
     try:
         order = Order.objects.get(id=ordernum)
-    except Order.DoesNotExist:
-        raise Http404('Order not found!')
+    except (Order.DoesNotExist, ValueError):
+        if is_json_request:
+            return _get_external_order(request, ordernum)
+        else:
+            raise Http404('Order not found!')
 
     if order.user != request.user or order.status not in ['purchased', 'refunded']:
         raise Http404('Order not found!')
 
-    if 'application/json' in request.META.get('HTTP_ACCEPT', ""):
+    if is_json_request:
         return _show_receipt_json(order)
     else:
         return _show_receipt_html(request, order)
 
 
+def _get_external_order(request, order_number):
+    """Get the order context from the external E-Commerce Service.
+
+    Get information about an order. This function makes a request to the E-Commerce Service to see if there is
+    order information that can be used to render a receipt for the user.
+
+    Args:
+        request (Request): The request for the the receipt.
+        order_number (str) : The order number.
+
+    Returns:
+        dict: A serializable dictionary of the receipt page context based on an order returned from the E-Commerce
+            Service.
+
+    """
+    try:
+        api = EcommerceAPI()
+        order_number, order_status, order_data = api.get_order(request.user, order_number)
+        billing = order_data.get('billing_address', {})
+        country = billing.get('country', {})
+
+        # In order to get the date this order was paid, we need to check for payment sources, and associated
+        # transactions.
+        payment_date = None
+        for source in order_data.get('sources', []):
+            for transaction in source.get('transactions', []):
+                payment_date = transaction['date_created']
```

Comment:
It's a little weird actually. The sources and transactions could theoretically have more than one date associated with a single order, however our current receipt only expects one. It wouldn't be relative in the short term, but it does mean I'm making an assumption about which date to present, in the case that there is more than one. Maybe it should be the earliest date, or the latest?